### PR TITLE
follow-up: remove provider label after thread activity

### PR DESCRIPTION
**Pull request by elie222**
# User description
Ensure follow-up labels are removed from the email provider when a thread no longer needs follow-up, even if no tracker rows are updated. This keeps provider label state in sync during outbound and inbound activity.

- Always attempt provider label removal in clearFollowUpLabel while keeping DB tracker updates scoped to unresolved trackers.
- Add and update unit tests for the zero-updated-tracker path and unresolved tracker filter behavior.

---

# Generated description

Below is a concise technical summary of the changes proposed in this PR:
Ensures email provider follow-up labels are removed even when no database tracker rows are updated, maintaining synchronization between the provider and the application state. Refactors <code>clearFollowUpLabel</code> to verify the existence of a <code>threadTracker</code> before proceeding with label removal and database updates.
<table><tr><th>Topic</th><th>Details</th><tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1724?tool=ast&topic=Tests+%26+Docs>Tests & Docs</a>
        </td><td>Update unit tests to cover scenarios with zero updated trackers and add guidance on Prisma retry usage in documentation.<details><summary>Modified files (2)</summary><ul><li>AGENTS.md</li>
<li>apps/web/utils/follow-up/labels.test.ts</li></ul></details><details><summary>Latest Contributors(1)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>gmail-Add-serverless-s...</td><td>February 26, 2026</td></tr></table></details></td></tr>
<tr><td><a href=https://baz.co/changes/elie222/inbox-zero/1724?tool=ast&topic=Follow-up+Logic>Follow-up Logic</a>
        </td><td>Refactor <code>clearFollowUpLabel</code> to decouple provider label removal from the database update count and improve error handling.<details><summary>Modified files (1)</summary><ul><li>apps/web/utils/follow-up/labels.ts</li></ul></details><details><summary>Latest Contributors(2)</summary><table><tr><th>User</th><th>Commit</th><th>Date</th></tr><tr><td>elie222</td><td>follow-up-reduce-polli...</td><td>February 25, 2026</td></tr>
<tr><td>joshwerner001@gmail.com</td><td>Follow-up-reminders</td><td>January 08, 2026</td></tr></table></details></td></tr></table>
This pull request is reviewed by Baz. Review like a pro on <a href=https://baz.co/changes/elie222/inbox-zero/1724?tool=ast>(Baz)</a>.

### CODE REVIEW

File: apps/web/utils/follow-up/labels.ts [hunk 1]
```diff
@@ -139,17 +147,59 @@ export async function clearFollowUpLabel({
...
 
+  try {
     await removeFollowUpLabel({ provider, threadId, logger });
 
-    logger.info("Removed follow-up label and cleared tracker", {
+    logger.info("Completed follow-up label removal check", {
       threadId,
+      clearedTrackerCount,
+      removalReason,
     });
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/follow-up/labels.ts]: Good catch. Updated clearFollowUpLabel so DB failures no longer force label removal fallback. If tracker state cannot be confirmed, we now skip provider label removal and only log the DB error. This avoids removing a user-applied label when ownership is unknown.

File: apps/web/utils/follow-up/labels.test.ts [hunk 2]
```diff
@@ -297,15 +297,47 @@ describe("clearFollowUpLabel", () => {
...
+      threadId: "thread-1",
+      provider: mockProvider,
+      logger,
+    });
+
+    expect(prisma.threadTracker.findFirst).toHaveBeenCalledWith({
+      where: {
+        emailAccountId: "account-1",
+        threadId: "thread-1",
+        followUpAppliedAt: { not: null },
```
[the remark this answers is not in the document]
Comment (elie222) [reply apps/web/utils/follow-up/labels.test.ts:335]: Addressed. The fallback tracker lookup now includes resolved: false, so only unresolved follow-up trackers can trigger provider label removal. Updated tests to assert this filter and to verify DB-error path now skips label removal.
